dCommon/Lxfml.cpp

```diff
@@ -128,3 +152,258 @@ Lxfml::Result Lxfml::NormalizePosition(const std::string_view data, const NiPoin
 	toReturn.center = newRootPos;
 	return toReturn;
 }
+
+// Deep-clone an XMLElement (attributes, text, and child elements) into a target document
+static tinyxml2::XMLElement* CloneElementDeep(const tinyxml2::XMLElement* src, tinyxml2::XMLDocument& dstDoc) {
+	if (!src) return nullptr;
+	auto* dst = dstDoc.NewElement(src->Name());
+
+	// copy attributes
+	for (const tinyxml2::XMLAttribute* attr = src->FirstAttribute(); attr; attr = attr->Next()) {
+		dst->SetAttribute(attr->Name(), attr->Value());
+	}
+
+	// copy children (elements and text)
+	for (const tinyxml2::XMLNode* child = src->FirstChild(); child; child = child->NextSibling()) {
+		if (const tinyxml2::XMLElement* childElem = child->ToElement()) {
+			dst->InsertEndChild(CloneElementDeep(childElem, dstDoc));
+		} else if (const tinyxml2::XMLText* txt = child->ToText()) {
+			auto* n = dstDoc.NewText(txt->Value());
+			dst->InsertEndChild(n);
+		} else if (const tinyxml2::XMLComment* c = child->ToComment()) {
+			auto* n = dstDoc.NewComment(c->Value());
+			dst->InsertEndChild(n);
+		}
+	}
+
+	return dst;
+}
+
+std::vector<Lxfml::Result> Lxfml::Split(const std::string_view data, const NiPoint3& curPosition) {
```

Comment:
How well does this function handle bad lxfml data?  it would be trivial to send a payload with bad data to the server, please check for error handling with either malformed xml, malformed _lxfml_ (i.e. an invalid lxfml format) and lxfml with extra data in it.  We know the client will likely only send good lxfml, but we can never guarantee user input is good.

Reply:
It did not handle it well, neither did parts of the normalization function, I have addressed issues in both functions, added tests for theses cases, and re-tested everything in-game